test/string_test.dart

```diff
@@ -9,6 +9,7 @@ import 'package:yaml_edit/yaml_edit.dart';
 final _testStrings = [
   "this is a fairly' long string with\nline breaks",
   'whitespace\n after line breaks',
+  'whitespace\n \nbetween line breaks',
```

Comment:
by the way, feel free to add lots of cases here.

I'd honestly suggest adding anything we think could cause a problem. 
```suggestion
  'whitespace\n \nbetween line breaks',
  'whitespace and line breaks at end 1\n ',
  'whitespace and line breaks at end 2 \n \n',
  'whitespace and line breaks at end 3 \n\n',
  'whitespace and line breaks at end 3 \n\n ',
```
If we find something that's still broken, we can just comment it out, and file a bug to fix it later.